src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -122,6 +122,7 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
                             // If an exception occurs, ensure the other awaiters
                             // render the output by themselves.
                             tcs.SetResult(null);
+                            tokenSource.Dispose();
```

Comment:
Does anything happen to `tokenSource.Token` or the `CancellationChangeToken` containing it when the source is disposed?